### PR TITLE
[Privatization] Add MakeOnlyUsedByChildrenProtectedRector

### LINKED ISSUE #5078: [Privatization] Make public class method protected, if only used by its children
E.g.

```diff
abstract class AbstractSomeClass
{
-	public function run()
+ 	protected function run()
    {
    }
}

class ChildClass extends AbstractSomeClass
{
	public function go()
    {
        $this->run();
    }
}
```

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes #5078

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by samsonasik** [comment]
I updated to skip extended overridden method

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/privatization/tests/Rector/ClassMethod/MakeOnlyUsedByChildrenProtectedRector/Fixture/skip_used_by_external_in_closure.php.inc [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    {
+    }
+}
+
+function () {
+    $obj = new SkipUsedByExternalInClosure();
+    $obj->run();
+};
+
+?>
```
Comment (TomasVotruba) [inline rules/privatization/tests/Rector/ClassMethod/MakeOnlyUsedByChildrenProtectedRector/Fixture/skip_used_by_external_in_closure.php.inc:17]: There should be also use case with parent class method, that has a child, and the child method call is used.

```php
abstract class SomeParent
{
    public function run()
    {
    }
}

final class ChildClass extends SomeParent
{
}

final class SkipSomeUser
{
     public function useIt(ChildClass $childClass)
     {
         // the method is used, so it should be skipped
         $childClass->run();
     }
}
```

**Comment by TomasVotruba** [comment]
Looks very good, just one detail :+1:

File: rules/privatization/tests/Rector/ClassMethod/MakeOnlyUsedByChildrenProtectedRector/Fixture/skip_used_by_external_in_closure.php.inc (hunk 1 above)
Comment (samsonasik) [reply rules/privatization/tests/Rector/ClassMethod/MakeOnlyUsedByChildrenProtectedRector/Fixture/skip_used_by_external_in_closure.php.inc:17]: implemented, added fixture for it.

**Comment by TomasVotruba** [comment]
Thanks :+1:

**Pull request merged**